Code/Tools/AssetProcessor/native/utilities/SimpleStatsCapture.cpp

```diff
@@ -0,0 +1,392 @@
+                {
+                    dest.push_back(element.first);
+                }
+            }
+            
+            // Make a friendly time string of the format nnHnnMhhS.xxxms
+            AZStd::string FormatDuration(const duration& duration)
+            {
+                int64_t milliseconds = duration.count();
+                constexpr int64_t millisecondsInASecond = 1000;
+                constexpr int64_t millisecondsInAMinute = millisecondsInASecond * 60;
+                constexpr int64_t millisecondsInAnHour = millisecondsInAMinute * 60;
+                
+                int64_t hours = milliseconds / millisecondsInAnHour;
+                milliseconds -= hours * millisecondsInAnHour;
+                
+                int64_t minutes = milliseconds / millisecondsInAMinute;
+                milliseconds -= minutes * millisecondsInAMinute;
+                
+                int64_t seconds = milliseconds / millisecondsInASecond;
+                milliseconds -= seconds * millisecondsInASecond;
+                
+                // omit the sections which dont make sense for readability
+                if (hours)
+                {
+                    return AZStd::string::format("%02" PRId64 "h%02" PRId64 "m%02" PRId64 "s%03" PRId64 "ms" , hours, minutes, seconds, milliseconds);
+                }
+                else if (minutes)
+                {
+                    return AZStd::string::format("   %02" PRId64 "m%02" PRId64 "s%03" PRId64 "ms", minutes, seconds, milliseconds);
+                }
+                else if (seconds)
+                {
+                    return AZStd::string::format("      %02" PRId64 "s%03" PRId64 "ms", seconds, milliseconds);
+                }
+                
+                return AZStd::string::format("         %03" PRId64 "ms", milliseconds);
+            }
+        };
+
+        void SimpleStatsCaptureImpl::BeginCaptureStat(AZStd::string_view statName)
+        {
+            StatsEntry& existingStat = m_stats[statName];
+            if (existingStat.m_operationStartTime != timepoint())
+            {
+                // prevent double 'Begins'
+                return;
+            }
+            existingStat.m_operationStartTime = AZStd::chrono::high_resolution_clock::now();
+        }
+
+        void SimpleStatsCaptureImpl::EndCaptureStat(AZStd::string_view statName)
+        {
+            StatsEntry& existingStat = m_stats[statName];
+            if (existingStat.m_operationStartTime != timepoint())
+            {
+                existingStat.m_cumulativeTime = AZStd::chrono::high_resolution_clock::now() - existingStat.m_operationStartTime;
+                existingStat.m_operationCount = existingStat.m_operationCount + 1;
+                existingStat.m_operationStartTime = timepoint(); // reset the start time so that double 'Ends' are ignored.
+            }
+        }
+
+        void SimpleStatsCaptureImpl::Dump()
```

Comment:
Yeah, the tests can also read stdout. There are some minor benefits to pulling this info from the communication port, but it's not a big deal either way, and like you said, it can always be added in the future.